### ISSUE TITLE
Comparing Offsets of Omega Words

### ISSUE THREAD

**Issue by fesemeyer** [leonbohn/lama#54]
Checking for equality of offsets of omega words (e.g. type `automata::word::Offset<'_, char, Reduced<char>>`) can return `false` even though the represented word is identical.
Failing code example:
```Rust
let w0 = Reduced::ultimately_periodic("a", "b");
let w1 = Reduced::periodic("b");
let offset0 = w0.offset(1);
let offset1 = w1.offset(0);
assert_eq!(offset0, offset1);
```
This behavior is probably fine and can be avoided by normalizing the offsets with `offset.normalized()`. However, I think there should be a method of the form `offset0.equals(offset1)` on offsets of omega words.
Note that this already exists for offsets of finite words.

**Comment by leonbohn** [comment]
Agreed! The implementation of `PartialEq`/`Eq` for these types of words should be revised. I'm not really sure whether it should be possible to have `assert_eq` fail even though the words are indeed equal.
What do you think, should the `Eq` implementation verify semantic equality, i.e. that the represented words are equal (in which case the behaviour on the provided snipped is problematic), or should it represent syntactic equality, which would mean we take the same offset of the same omega-word. For me, the second case does not really make too much sense and I cannot think of many situations in which we would be interested in the syntactic equality over the semantic equality.

**Comment by fesemeyer** [comment]
I also can't think of use cases that require syntactic equality but for this type I wouldn't necessarily expect `==` to work semantically. When I was looking for a solution I looked for an `equals` function first. Implementing this would probably be easier and still pretty intuitive in my opinion.
